// @okta/okta-angular/__ivy_ngcc__/dist/okta/okta.module.js
"use strict";
/*
 * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var components_1 = require("./components/");
var okta_service_1 = require("./services/okta.service");
var okta_guard_1 = require("./okta.guard");
var okta_config_1 = require("./models/okta.config");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./components/callback.component');
var ɵngcc2 = require('./components/login-redirect.component');
var OktaAuthModule = /** @class */ (function () {
    function OktaAuthModule() {
    }
    OktaAuthModule.initAuth = function (config) {
        return {
            ngModule: OktaAuthModule,
            providers: [
                okta_guard_1.OktaAuthGuard,
                okta_service_1.OktaAuthService,
                { provide: okta_config_1.OKTA_CONFIG, useValue: config }
            ]
        };
    };
    /** @nocollapse */
    OktaAuthModule.ctorParameters = function () { return []; };
OktaAuthModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: OktaAuthModule });
OktaAuthModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function OktaAuthModule_Factory(t) { return new (t || OktaAuthModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(OktaAuthModule, { declarations: [ɵngcc1.OktaCallbackComponent, ɵngcc2.OktaLoginRedirectComponent], exports: [ɵngcc1.OktaCallbackComponent, ɵngcc2.OktaLoginRedirectComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OktaAuthModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [
                    components_1.OktaCallbackComponent,
                    components_1.OktaLoginRedirectComponent
                ],
                exports: [
                    components_1.OktaCallbackComponent,
                    components_1.OktaLoginRedirectComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return OktaAuthModule;
}());
exports.OktaAuthModule = OktaAuthModule;
