// @okta/okta-angular/__ivy_ngcc__/dist/okta/components/login-redirect.component.js
"use strict";
/*
 * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var okta_service_1 = require("../services/okta.service");
var ɵngcc0 = require('@angular/core');
var OktaLoginRedirectComponent = /** @class */ (function () {
    function OktaLoginRedirectComponent(okta) {
        this.okta = okta;
        okta.loginRedirect();
    }
    /** @nocollapse */
    OktaLoginRedirectComponent.ctorParameters = function () { return [
        { type: okta_service_1.OktaAuthService, },
    ]; };
OktaLoginRedirectComponent.ɵfac = function OktaLoginRedirectComponent_Factory(t) { return new (t || OktaLoginRedirectComponent)(ɵngcc0.ɵɵdirectiveInject(okta_service_1.OktaAuthService)); };
OktaLoginRedirectComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: OktaLoginRedirectComponent, selectors: [["ng-component"]], decls: 0, vars: 0, template: function OktaLoginRedirectComponent_Template(rf, ctx) { }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OktaLoginRedirectComponent, [{
        type: core_1.Component,
        args: [{ template: "" }]
    }], function () { return [{ type: okta_service_1.OktaAuthService }]; }, null); })();
    return OktaLoginRedirectComponent;
}());
exports.OktaLoginRedirectComponent = OktaLoginRedirectComponent;
